docs/reference/builder.md

```diff
@@ -1201,6 +1207,19 @@ the variable's value in the `ENV` references the `ARG` variable and that
 variable is changed through the command line. In this example, the `ENV`
 command causes the image to include the value.
 
+If an `ENV` instruction overrides an `ARG` instruction of the same name, like
+this Dockerfile:
+
+```
+1 FROM ubuntu
+2 ARG CONT_IMG_VER
+3 ENV CONT_IMG_VER hello
+4 RUN echo $CONT_IMG_VER
+```
+There will be no cache miss on line 3 because the value of `CONT_IMG_VER` is now
```

Comment:
Small change.  @marsmensch Let us know if you need a carry to finish this out.

```markdown
Line 3 does not cause a cache miss because the value of `CONT_IMG_VER` is a
constant (`hello`). As a result, the environment variables and values used on
the `RUN` (line 4) doesn't change between builds.
```